tools/cache_creator/cache_creator.cpp

```diff
@@ -161,6 +162,29 @@ bool hexStringToUuid(llvm::StringRef hexStr, llvm::MutableArrayRef<uint8_t> outU
   return true;
 }
 
+llvm::Expected<ElfLlpcCacheInfo> getCacheInfoFromMetadataBlob(llvm::StringRef noteBlob) {
+  llvm::msgpack::Document *document = new llvm::msgpack::Document;
+  if (!document->readFromBlob(noteBlob, false)) {
+    return llvm::createStringError(llvm::object::object_error::invalid_file_type, "Bad PAL metadata format");
+  }
+
+  llvm::msgpack::MapDocNode pipelineNode =
+      document->getRoot().getMap(true)["amdpal.pipelines"].getArray(true)[0].getMap(true);
+  llvm::msgpack::MapDocNode cacheInfoNode = pipelineNode[".xgl_cache_info"].getMap(true);
```

Comment:
Why does convert have to be set to `true`? Could this be hiding potential bugs in the code?

Reply:
It doesn't need to be `true`, fixed up in the next version to be more more explicit with expected format.